bcgov/PIMS
# PIMS-2107 Email Field Rework

## 🎯 Summary

  
[PIMS-2107](https://citz-imb.atlassian.net/browse/PIMS-2107)


## Changes
- API now verifies emails when submitting an access request or updating a user.
- Email field on the Access Request form is now only prepopulated and locked when it's an IDIR user
- Non-IDIR users can enter emails of their choosing
- Frontend uses Zod to validate the email to stay consistent with the backend.
- User Details editing also has validation added.

## Testing
- Log in with IDIR and BCSC user. See how email field is restricted or not
- Try entering a bad email and submitting
- Try entering a good email and submitting


## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
